utils/writeable/src/cmp.rs

```diff
@@ -19,7 +19,8 @@ impl fmt::Write for WriteComparator<'_> {
             return Ok(());
         }
         let cmp_len = core::cmp::min(other.len(), self.code_units.len());
-        let (this, remainder) = self.code_units.split_at(cmp_len);
+        // SAFETY: cmp_len is at most self.code_units.len()
+        let (this, remainder) = unsafe { self.code_units.split_at_unchecked(cmp_len) };
```

Comment:
No need for unsafe here

```suggestion
        let (this, remainder) = self.code_units.split_at_checked(other.len())
            .unwrap_or((self.code_units, &[]));
```

Reply:
This is a very hot code path. `split_at_checked` introduces a bounds check that might not be optimized out.

Reply:
We got by with the bounds check so far and had no apparent performance issues. I don't think we should introduce unsafe unless we have actual numbers on what this affects, and whether optimizations are indeed inhibited.

Reply:
Oh I didn't notice that. This reduces the number of bounds checks back to what we had before.